### ISSUE TITLE
EnumConverter doesn't work when fromType is Kotlin Int

### ISSUE THREAD

**Issue by oripwk** [jOOQ/jOOQ#8045]
### Expected behavior and actual behavior:
When defining a field like this:
```kotlin
val STATE: Field<MyEnum> = DSL.field("my_enum", SQLDataType.INTEGER.asConvertedDataType(EnumConverter<Int, MyEnum>(Int::class.java, MyEnum::class.java)))
```
And then reading it:
```kotlin
jooq.selectFrom(DSL.table("example")).fetchInto(Example::class.java)
```

jOOQ will return `null` for `my_enum`. The reason it happens, is that in `EnumConverter` it decides if it's an ordinal or string converter, and it mistakenly decides it's a string converter since the conditional:

https://github.com/jOOQ/jOOQ/blob/5ff097fcbd1646802092038951c446bbb66a3c7c/jOOQ/src/main/java/org/jooq/impl/EnumConverter.java#L63

won't work.

**This happens since Kotlin's `Int` type is not assignable into `java.lang.Number`**

### Steps to reproduce the problem (include example code if possible):
As I described above.
### Versions:

- jOOQ: 3.11.7
- Java: 1.8
- Database (include vendor): MySQL 5.7
- OS: MacOS
- JDBC Driver (include name if inofficial driver): mysql-connector-java 6.0.6

**Comment by lukaseder** [comment]
Thank you very much for your report

> This happens since Kotlin's Int type is not assignable into java.lang.Number

This opens up a variety of issues throughout jOOQ, not just in the `EnumConverter`. For the record, Kotlin has its own `Number` base type:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-number/index.html

I will collect other Kotlin interop issues before moving on with this one. The workaround for this issue is easy: Create your own converter

**Comment by lukaseder** [comment]
Looking into this now in the context of https://github.com/jOOQ/jOOQ/issues/6248

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10261)
